test/components/dragBoxLayerTests.ts

```diff
@@ -162,22 +174,95 @@ describe("Interactive Components", () => {
+      var callbackCalled = false;
       var callback = (b: Plottable.Bounds) => {
         receivedBounds = b;
+        callbackCalled = true;
       };
       dbl.onDragEnd(callback);
 
       var target = dbl.background();
       TestMethods.triggerFakeDragSequence(target, startPoint, endPoint);
 
+      assert.isTrue(callbackCalled, "the callback was called");
       assert.deepEqual(receivedBounds.topLeft, startPoint, "top-left point was set correctly");
       assert.deepEqual(receivedBounds.bottomRight, endPoint, "bottom-right point was set correctly");
-      assert.strictEqual(dbl.onDragEnd(), callback, "can retrieve callback by calling with no args");
-      dbl.onDragEnd(null);
-      assert.isNull(dbl.onDragEnd(), "can blank callback by passing null");
+      dbl.offDragEnd(callback);
+      callbackCalled = false;
+
+      TestMethods.triggerFakeDragSequence(target, startPoint, endPoint);
+      assert.isFalse(callbackCalled, "the callback was detached from the dragoBoxLayer and not called");
+
+      svg.remove();
+    });
+
+    it("multiple drag interaction callbacks", () => {
+      var svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var dbl = new Plottable.Components.DragBoxLayer();
+      dbl.renderTo(svg);
+
+      var startPoint = {
+        x: SVG_WIDTH / 4,
+        y: SVG_HEIGHT / 4
+      };
+      var endPoint = {
+        x: SVG_WIDTH / 2,
+        y: SVG_HEIGHT / 2
+      };
+
+      var callbackDragStart1Called = false;
+      var callbackDragStart2Called = false;
+      var callbackDrag1Called = false;
+      var callbackDrag2Called = false;
+      var callbackDragEnd1Called = false;
+      var callbackDragEnd2Called = false;
+
+      var callbackDragStart1 = () => callbackDragStart1Called = true;
+      var callbackDragStart2 = () => callbackDragStart2Called = true;
+      var callbackDrag1 = () => callbackDrag1Called = true;
+      var callbackDrag2 = () => callbackDrag2Called = true;
+      var callbackDragEnd1 = () => callbackDragEnd1Called = true;
+      var callbackDragEnd2 = () => callbackDragEnd2Called = true;
+
+      dbl.onDragStart(callbackDragStart1);
+      dbl.onDragStart(callbackDragStart2);
+      dbl.onDrag(callbackDrag1);
+      dbl.onDrag(callbackDrag2);
+      dbl.onDragEnd(callbackDragEnd1);
+      dbl.onDragEnd(callbackDragEnd2);
+
+      var target = dbl.background();
+      TestMethods.triggerFakeDragSequence(target, startPoint, endPoint);
+
+      assert.isTrue(callbackDragStart1Called, "the callback 1 for drag start was called");
+      assert.isTrue(callbackDragStart2Called, "the callback 2 for drag start was called");
+      assert.isTrue(callbackDrag1Called, "the callback 1 for drag was called");
+      assert.isTrue(callbackDrag2Called, "the callback 2 for drag was called");
+      assert.isTrue(callbackDragEnd1Called, "the callback 1 for drag end was called");
+      assert.isTrue(callbackDragEnd2Called, "the callback 2 for drag end was called");
```

Comment:
Can we move these assert statements so that they are right after their corresponding action?

Like
```typescript
dbl.onDragStart...
assert.isTrue(callbackDragStart1Called...
...
```